Libraries/Image/RCTImageLoader.m

```diff
@@ -400,7 +400,13 @@ - (RCTImageLoaderCancellationBlock)loadImageOrDataWithURLRequest:(NSURLRequest *
             return;
           }
 
-          NSURL *redirectURL = [NSURL URLWithString: location];
+          NSURL *redirectURL;
+          if (![location hasPrefix:@"http"]) {
```

Comment:
@nicklockwood That works. I was afraid it wouldn't work when the redirect URL has a different domain, but it's smart enough.